### DISCUSSION TITLE
Trying to create a HighWay Env with an Emergency Vehicle

### DISCUSSION THREAD

**Discussion by malhardamle** [eleurent/highway-env#458] [Q&A]
Hi all, 
I am trying to add a new emergency vehicle to the highway env. I want to add an ambulance, firetruck, and police car vehicles to the current highway env.  I cannot find documentation regarding the vehicle graphics class. Any advice on how to go about adding these emergency vehicles? Ex. I want to create a pink car (police car) moving faster than the agent (green vehicle), an orange car( ambulance) moving faster than the agent car, etc. 

What would be the best way to add a new vehicle with the color nad speed specifications to an existing high way env?

IM using jupyter notebook to run the scripts and code. Currently, my pygame graphics open for a couple of seconds after building and running the highway env. After a couple of seconds, the external py game graphics window freezes and stops responding. Any ideas on why this happens?

**Comment by eleurent** [comment]
> What would be the best way to add a new vehicle with the color nad speed specifications to an existing high way env?

Whenever you create a vehicle, you can manually set their `color` property to any rgba color, see e.g. this line:
https://github.com/Farama-Foundation/HighwayEnv/blob/0a174bef8b9418c8e7fea80d662b017d4a83cb31/highway_env/envs/parking_env.py#L155

You can also set their initial speed with `vehicle.speed` and target speed with `vehicle.target_speed`.

> IM using jupyter notebook to run the scripts and code. Currently, my pygame graphics open for a couple of seconds after building and running the highway env. After a couple of seconds, the external py game graphics window freezes and stops responding. Any ideas on why this happens?

I'm not sure why this is happening. Maybe make sure that everytime that you enable rendering to generate a video, the pygame window is properly closed by calling `env.close()` ?
